fxa/oauth.py

```diff
@@ -177,3 +178,55 @@ def verify_token(self, token, scope=None):
                 raise ScopeMismatchError(authorized_scope, scope)
 
         return resp
+
+
+class MemoryCache(object):
+    """Simple Memory cache that caches the object in the memory."""
+
+    def __init__(self, ttl):
+        self.ttl = ttl
+        if not self.ttl:
+            self.ttl = DEFAULT_CACHE_EXPIRACY
+        self.cache = {}
+        self.expires_at = {}
+
+    def get(self, key):
+        value = self.cache.get(key, None)
+        expires_at = self.expires_at(key, None)
+        if expires_at and expires_at > time.time():
+            return value
+        else:
+            self.delete(key)
+
+    def set(self, key, value):
+        self.cache[key] = value
+        self.expires_at[key] = time.time() + self.ttl
```

Comment:
When setting the value (or maybe when getting it?) we should have a look at all the keys and expire the ones which are here since too long, otherwise memory will probably grow out too quickly.